preview-maker.js

```diff
@@ -1,184 +1,376 @@
+  }
+  let colors = [];
+  for (const variable in defaultVariablesValues) {
+    if (variable === `chat_wallpaper` && !theme.chat_wallpaper) {
+      continue;
+    }
+    const elements = getElementsByClassName(preview, variable);
+    const color = theme[variable] || defaultVariablesValues[variable];
+    for (const element of elements) {
+      fill(element, color);
+    }
+    const windowBackgroundWhite = rgbToHsl(
+      theme[`windowBackgroundWhite`] ||
+        defaultVariablesValues[`windowBackgroundWhite`]
+    );
+    const chooseHsl = rgbToHsl(color);
+    let hueDifference = Math.abs(chooseHsl.hue - windowBackgroundWhite.hue);
+    if (hueDifference > 180) {
+      hueDifference = 360 - hueDifference;
+    }
+    if (hueDifference > 6 && chooseHsl.saturation > 0.08) {
+      colors.push(Math.round(chooseHsl.hue));
+    }
+  }
+
+  if (colors.length) {
+    accentHue = calculateAccentColor(colors);
+  } else {
+    accentHue = rgbToHsl(
+      theme[`chats_actionBackground`] ||
+        defaultVariablesValues[`chats_actionBackground`]
+    ).hue;
+  }
+
+  for (const outBubbleGradientelement of getElementsByClassName(
+    preview,
+    `chat_outBubbleGradient`
+  )) {
+    const color =
+      theme[`chat_outBubbleGradient`] ||
+      theme[`chat_outBubble`] ||
+      defaultVariablesValues[`chat_outBubble`];
+    fill(outBubbleGradientelement, color);
+  }
+  for (const previewBackLinear of getElementsByClassName(
+    preview,
+    `PreviewBackLinear`
+  )) {
+    const chatOutBubble =
+      theme[`chat_outBubble`] || defaultVariablesValues[`chat_outBubble`];
+    fill(previewBackLinear, {
+      hue: rgbToHsl(theme[`chat_outBubbleGradient`] || chatOutBubble).hue,
+      saturation: 1,
+      lightness: 0.9,
+    });
+  }
+  for (const previewBackLinearShadow of getElementsByClassName(
+    preview,
+    `PreviewBackLinearShadow`
+  )) {
+    fill(previewBackLinearShadow, {
+      hue: rgbToHsl(
+        theme[`chat_outBubble`] || defaultVariablesValues[`chat_outBubble`]
+      ).hue,
+      saturation: 1,
+      lightness: 0.9,
+    });
+  }
+  for (const previewBack of getElementsByClassName(preview, `PreviewBack`)) {
+    fill(previewBack, { hue: accentHue, saturation: 1, lightness: 0.9 });
+  }
+  for (const chatShadow of getElementsByClassName(preview, `ChatShadow`)) {
+    fill(chatShadow, { hue: accentHue, saturation: 1, lightness: 0.02 });
+  }
+  if (`chat_outBubbleGradient` in theme) {
+    for (const PreviewBack of getElementsByClassName(preview, `PreviewBack`)) {
+      fill(PreviewBack, { red: 0, green: 0, blue: 0, alpha: 0 });
+    }
+  }
+
+  for (const avatar of AVATAR_VARIABLES) {
+    const windowBackgroundWhite =
+      theme[`windowBackgroundWhite`] ||
+      defaultVariablesValues[`windowBackgroundWhite`];
+    const avatarColor = theme[avatar] || defaultVariablesValues[avatar];
+    const avaAndBack = rgbDifference(avatarColor, windowBackgroundWhite);
+    const avaTextColor =
```

Comment:
```suggestion
    const avatarTextColor =
```

This variable can also be moved outside the loop.